### PR TITLE
add ai cv-analyzer implementation

### PR DESCRIPTION
**Pull request by bluel1ner**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -20,11 +20,14 @@ repositories {
 
 dependencies {
 	implementation("org.springframework.boot:spring-boot-starter-web")
-	implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
 	implementation("org.jetbrains.kotlin:kotlin-reflect")
 	testImplementation("org.springframework.boot:spring-boot-starter-test")
 	testImplementation("org.jetbrains.kotlin:kotlin-test-junit5")
 	testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+	implementation("com.fasterxml.jackson.module:jackson-module-kotlin:2.15.0")
+	compileOnly("org.projectlombok:lombok:1.18.30")
```
Comment (yahorbarkouski) [inline build.gradle.kts]: you don't need lombok in kotlin :smile:

File: src/main/kotlin/org/blueliner/springbootwithkotlin/dto/request/OpenAiRequest.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.blueliner.springbootwithkotlin.dto.request
+
+data class OpenAiRequest (
+    val model: String,
+    val max_tokens: Int,
```
Comment (yahorbarkouski) [inline src/main/kotlin/org/blueliner/springbootwithkotlin/dto/request/OpenAiRequest.kt]: use json camel case serialization strategy, underscores in field namings are bad practice

File: src/main/kotlin/org/blueliner/springbootwithkotlin/dto/response/CvScoringBySectionResponse.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package org.blueliner.springbootwithkotlin.dto.response
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import lombok.NoArgsConstructor
+
+@NoArgsConstructor
```
Comment (yahorbarkouski) [inline src/main/kotlin/org/blueliner/springbootwithkotlin/dto/response/CvScoringBySectionResponse.kt]: redundant, data class = @Data in lombok by default

File: src/main/kotlin/org/blueliner/springbootwithkotlin/service/CvScoringService.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+class CvScoringService {
+    val objectMapper = ObjectMapper()
+
+    @Autowired
+    lateinit var aiClient: AiClientService
+
+    fun getCommonCvScoring(file: MultipartFile): CvScoringResponse {
+        val pdfPayload = PdfUtils.parsePdfToString(file)
+        val message: String =
+            "Hey! I have a cv and you should score it by range from 0 to 100 and respond in json" +
```
Comment (yahorbarkouski) [inline src/main/kotlin/org/blueliner/springbootwithkotlin/service/CvScoringService.kt]: you don't need to greet ai:) the less tokens you use the better

**Review by yahorbarkouski — APPROVED** [review]
lgtm overall please remove lombok!!!!

File: src/main/kotlin/org/blueliner/springbootwithkotlin/service/AiClientService.kt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+
+    private fun getHeaders(): HttpHeaders {
+        return HttpHeaders().apply {
+            contentType = MediaType.APPLICATION_JSON
+            set(HttpHeaders.AUTHORIZATION, BEARER_TOKEN.plus(openAiKey))
+        }
+    }
+
+    private fun getBody(message: String) = OpenAiRequest(
```
Comment (bluel1ner) [inline src/main/kotlin/org/blueliner/springbootwithkotlin/service/AiClientService.kt:43]: Here i use a simple json_object but it doesn't guarantee getting the body i want from ai

The main question is how to pass json_schema to response_format via sdk objects not manual https://platform.openai.com/docs/guides/structured-outputs/examples?context=with_parse

**Pull request merged**